src/schema/README.md

```diff
@@ -0,0 +1,36 @@
+# BIDS-schema
+
+Portions of the BIDS specification are defined using YAML files, in order to
+make the specification machine-readable.
+
+Currently, the only portion of the specification that relies on this schema is
+the Entity Table, but any changes to the specification should be mirrored in the
+schema.
```

Comment:
Not just the entity table, but also the entity definitions appendix and filename patterns throughout the specification.

Reply:
I was trying to see if I can do a pull request about [this suggestion](https://github.com/bids-standard/bids-specification/pull/736#issuecomment-781194146) I made for renaming the `entity` field (to `key`) in the `entity.yaml` file. But I have a hard time to find all the places that would be affected by this change, can you perhaps tell me, how do I test this?

Reply:
I suspect most of the changes to be done there will be in the `tools/schema_code`, right @tsalo ?

Should we add a README into the tools folder with at least some very brief explanation of what to find in there?

Reply:
I agree that the code in that tools folder is where you will get the best idea of what parts of the schema are actively in use. It would be useful to add a README to that folder for right now, but I think we'll be moving that code into its own repository in the long run.

An alternative would be to focus on the functions that are directly used in the specification (i.e., [`macros.py`](https://github.com/bids-standard/bids-specification/blob/c8f5827d549d519ca9ebb0d3eebf6661ff348484/tools/schemacode/macros.py)).

Reply:
OK I can try checking the code base and improving the doc strings.